### ISSUE TITLE
[INSTALL]: MET/11.1.1

### ISSUE THREAD

**Issue by mkavulich** [JCSDA/spack-stack#1187]
### Package name

MET

### Package version/tag

11.1.1

### Build options

current

### Installation timeframe

Please install on Hera for testing then include in next release. We would prefer this be done by the end of the month if possible; I hope that's not too big an ask for a minor upgrade.

### Other information

This is a minor bugfix release of MET ([release page](https://github.com/dtcenter/MET/releases/tag/v11.1.1)), and it contains two specific bug fixes that we need for new smoke verification capabilities we are adding to the UFS SRW App:

 - https://github.com/dtcenter/MET/issues/2782
 - https://github.com/dtcenter/MET/pull/2784

**Comment by RatkoVasic-NOAA** [comment]
@mkavulich which spack stack version on Hera are you suggesting to install to? And under which environment? Thanks.

**Comment by RatkoVasic-NOAA** [comment]
@mkavulich can you please try installation of the met@11.1.1 using this way on Hera:
```
module purge
module use /scratch2/NCEPDEV/stmp1/role.epic/installs/gnu/modulefiles
module use /scratch2/NCEPDEV/stmp1/role.epic/installs/openmpi/modulefiles
module use /scratch2/NCEPDEV/stmp1/role.epic/spack-stack/spack-stack-1.6.0_gnu13/envs/met-unified-env/install/modulefiles/Core
module load stack-gcc
module load stack-openmpi
module load met
module list
```

**Comment by mkavulich** [comment]
@RatkoVasic-NOAA sorry, I missed your previous message. I have tested out this build on Hera and it appears to be working as expected. Thank you for your work!

**Comment by climbfuji** [comment]
PR https://github.com/JCSDA/spack-stack/pull/1228 was just merged - this updates met to 11.1.1 in the next spack-stack release. It's also available on Hera in spack-stack-1.6.0 for testing.

**Comment by mkavulich** [comment]
@RatkoVasic-NOAA It looks like the MET upgrade to 11.1.1 was not carried over to the latest build used for the UFS weather model (the builds named `upp-addon-env`). I assume this was an unintentional omission? I find it hard to follow the conventions for these non-numbered builds, but it looks like this environment was built more than a month after the MET upgrade was merged.

**Comment by RatkoVasic-NOAA** [comment]
@mkavulich MET was upgraded from 11.1.0 to 11.1.1 in spack-stack 1.8.0 (as you asked in this issue).
In 1.6.0 we made only one chained environment for you to test (on Hera and only GNU compiler). [Here](https://github.com/JCSDA/spack-stack/issues/1187#issuecomment-2276795024).
I guess you need met@11.1.1 in 1.6.0/fms-2024.01 (latest chained environment used by UFS WM) and/or 1.6.0/upp-addon-env (used by SRW)?
One solution is to add new version met@11.1.1 to existing met@11.1.0 in chained environment. That would be possible, but we already have metplus@5.1.0 depending on met@11.1.0 (you are using metplus, right?).

**Comment by mkavulich** [comment]
Yes, we are using metplus, but the additional fix we need is through MET. From what I gather, the dependency you describe means we couldn't load met 11.1.1 with metplus 5.1.0? Would it be possible to make the dependency conditional (i.e. metplus 5.1.0 requires either met@11.1.0 or met@11.1.1)? I know this can work with lua modules but I am not sure if this would be possible with spack-stack directly.

**Comment by mkavulich** [comment]
Alternatively, met 11.1.1 should be entirely back-compatible with met 11.1.0 as it is a simple bugfix patch, so updating the dependency shouldn't have any negative effects on existing users

**Comment by RatkoVasic-NOAA** [comment]
@AlexanderRichert-NOAA can you please chime in with suggestion how to add/replace met@11.1.0 with met@11.1.1 on chained environments that are already in use?
@mkavulich do you need it only for SRW?

**Comment by mkavulich** [comment]
Yes, this is just needed for new SRW verification capabilities.

**Comment by AlexanderRichert-NOAA** [comment]
The safe bet is to create a new chained environment. That said, if no other teams use met/metplus in upp-addon-env and @mkavulich and team don't mind the risk involved in modifying an existing environment, then you could modify upp-addon-env in place.

**Comment by RatkoVasic-NOAA** [comment]
I tested it on Hercules. It looks OK.
I'll go ahead and install met@11.1.1 and point metplus@5.1.0 to met@11.1.1 under 1.6.0/upp-addon-env/ on all machines.
